### ISSUE TITLE
Pacin в быстрой сортировке

### ISSUE THREAD

**Issue by KazikovAP** [Sparkmoons/go-ez-tasks#2]
Также при запуске решения быстрой сортировки выдаёт панику 

Возможное решение 

func quickSort(a []int) []int {
	if len(a) <= 1 {
		return a
	}

	min := make([]int, 0)
	max := make([]int, 0)
	res := make([]int, 0)

	pivot := a[len(a)/2]

	for _, num := range a {
		if num < pivot {
			min = append(min, num)
		} else if num > pivot {
			max = append(max, num)
		}
	}

	res = append(res, quickSort(min)...)
	for _, num := range a {
		if num == pivot {
			res = append(res, num)
		}
	}
	res = append(res, quickSort(max)...)

	return res
}

**Comment by Sparkmoons** [comment]
Да, условие выхода из рекурсии добавить забыл и в конце косяк. Поправил.02.06.2024, 16:27, "Aleksey Kazikov" ***@***.***>:
Также при запуске решения быстрой сортировки выдаёт панику
Возможное решение
func quickSort(a []int) []int {
if len(a) <= 1 {
return a
}
min := make([]int, 0)
max := make([]int, 0)
res := make([]int, 0)

pivot := a[len(a)/2]

for _, num := range a {
	if num < pivot {
		min = append(min, num)
	} else if num > pivot {
		max = append(max, num)
	}
}

res = append(res, quickSort(min)...)
for _, num := range a {
	if num == pivot {
		res = append(res, num)
	}
}
res = append(res, quickSort(max)...)

return res

}
